src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/court/libra/LibraCourtCaseListenerIntTest.kt

```diff
@@ -0,0 +1,147 @@
+
+  @BeforeEach
+  override fun beforeEach() {
+    super.beforeEach()
+    personRepository.deleteAll()
+    telemetryRepository.deleteAll()
+  }
+
+  @Test
+  fun `should process libra messages`() {
+    val messageId = publishHMCTSMessage(libraHearing(), LIBRA_COURT_CASE)
+
+    checkTelemetry(
+      HMCTS_MESSAGE_RECEIVED,
+      mapOf(
+        "EVENT_TYPE" to LIBRA_COURT_CASE.name,
+        "MESSAGE_ID" to messageId,
+      ),
+    )
+
+    val personEntities = await.atMost(30, SECONDS) untilNotNull {
+      personRepository.findAll()
+    }
+
+    assertThat(personEntities.size).isEqualTo(1)
+
+    val person = personEntities[0]
+    assertThat(person.title).isEqualTo("Mr")
+    assertThat(person.lastName).isEqualTo("MORGAN")
+    assertThat(person.pnc).isEqualTo(PNCIdentifier.from("2003/0011985X"))
+    assertThat(person.cro).isEqualTo(CROIdentifier.from("85227/65L"))
+    assertThat(person.dateOfBirth).isEqualTo(LocalDate.of(1975, 1, 1))
+    assertThat(person.addresses.size).isEqualTo(1)
+    assertThat(person.addresses[0].postcode).isEqualTo("NT4 6YH")
+  }
+
+  @Test
+  fun `should process send correct telemetry for candidate search`() {
+    personRepository.saveAndFlush(
+      PersonEntity.from(
+        Person(
+          firstName = "Jane",
+          lastName = "Smith",
+          addresses = listOf(Address(postcode = "LS1 1AB")),
+          sourceSystemType = SourceSystemType.HMCTS,
+        ),
+      ),
+    )
+    personRepository.saveAndFlush(
+      PersonEntity.from(
+        Person(
+          firstName = "Steve",
+          lastName = "Micheal",
+          addresses = listOf(Address(postcode = "RF14 5DG")),
+          sourceSystemType = SourceSystemType.HMCTS,
+        ),
+      ),
+    )
+
+    await untilAsserted { assertThat(personRepository.findAll().size).isEqualTo(2) }
+
+    publishHMCTSMessage(
+      libraHearing(
+        firstName = "Jane",
+        lastName = "Smythe",
+        postcode = "LS1 1AB",
+      ),
+      LIBRA_COURT_CASE,
+    )
+
+    checkTelemetry(
+      TelemetryEventType.CPR_CANDIDATE_RECORD_SEARCH,
+      mapOf(
+        "SOURCE_SYSTEM" to "HMCTS",
+        "EVENT_TYPE" to LIBRA_COURT_CASE.name,
+        "RECORD_COUNT" to "1",
+      ),
+    )
+  }
+
+  @Test
+  fun `should process concurrent libra messages with data reads`() {
+    val blitzer = Blitzer(50, 5)
+    try {
+      blitzer.blitz {
+        courtCaseEventsTopic?.snsClient?.publish(buildPublishRequest())?.get()
+      }
+    } finally {
+      blitzer.shutdown()
+    }
+
+    await untilCallTo {
+      courtCaseEventsQueue?.sqsClient?.countMessagesOnQueue(courtCaseEventsQueue!!.queueUrl)?.get()
+    } matches { it == 0 }
+
+    await untilCallTo {
+      courtCaseEventsQueue?.sqsDlqClient?.countMessagesOnQueue(courtCaseEventsQueue!!.dlqUrl!!)?.get()
+    } matches { it == 0 }
+
+    checkTelemetry(
+      CPR_RECORD_CREATED,
+      mapOf("SOURCE_SYSTEM" to "HMCTS"),
+      times = 50,
```

Comment:
Would have to pull out every sqs message id and search for each one, if we didn't wipe the telemetry repo.